src/application-manager.coffee

```diff
@@ -808,10 +807,12 @@ module.exports = class ApplicationManager extends EventEmitter
 			_.map app.services, (service) ->
 				img = _.find(available, { dockerImageId: service.config.image, imageId: service.imageId }) ? _.find(available, { dockerImageId: service.config.image })
 				return _.omit(img, [ 'dockerImageId', 'id' ])
+		allImageDockerIdsForTargetApp = (app) -> _.filter(_.map(app.services, (svc) -> [ svc.imageName, svc.config.image] ), (img) -> !_.isNull(img[1]))
```

Comment:
Also `img[1] != null` would be a bit clearer - does null vs undefined matter or was the `!== null` accidental?

Reply:
Done